api/src/models/JobMetadata.ts

```diff
@@ -83,30 +98,33 @@ class JobMetadata {
         this._documents = DocumentOrder.fromRaw(data);
     }
 
-    get raw() {
-        return {
-            order: this.order.raw,
-            published: this.published  
-        };
+    get audio() {
+        if (this._audio === null) {
+            this._audio = AudioOrder.fromJob(this.job);
+        }
+        return this._audio;
     }
 
-    set raw(data) {
-        //TODO: set raw data
+    set audio(data) {
+        this._audio = AudioOrder.fromRaw(data);
     }
 
     save() {
-
+        let fs = require('fs');
+        this._filename = this.job.dir + '/job.json'
```

Comment:
The constructor is building this same filename string in a local variable. Since you can guarantee that the constructor is always called first, I'd suggest moving this assignment into the constructor, using `this._filename` there in place of the local `filename` variable... and then you don't need to recalculate the filename here, since it will already be set.

Reply:
Updated, thanks!

Reply:
I see that you moved the assignment to `this._filename` into the constructor, but you didn't replace the redundant local `filename` variable. I'd recommend doing that to simplify everything.